qiskit/providers/aer/backends/aerbackend.py

```diff
@@ -182,6 +182,11 @@ def run(self,
                         run_options[key] = value
             qobj = self._assemble(circuits, **run_options)
         else:
+            noise_model = run_options.get("noise_model", self.options.get("noise_model"))
+            if noise_model:
```

Comment:
This should be built into either the `_assemble` or  `_assemble_noise_model` function, not handled as a special case outside it. The result of calling `qobj = backend._assemble(circuits)` should be exactly the qobj that is run.

My suggestion would to modify `_assemble_noise_model` like:
```python
        updated_noise = False
        noise_model = run_options.get(
            'noise_model', getattr(self.options, 'noise_model', None))
        
        # Add custom pass noise
        if noise_model:
            npm = noise_model._pass_manager()
            if npm is not None:
                circuits = npm.run(circuits)
         
        # Check if circuits contain quantum error instructions
        run_circuits = []
        # .. rest of code...

```

Reply:
631108d9